src/ustreamer/device.c

```diff
@@ -485,11 +500,13 @@ static int _device_open_check_cap(us_device_s *dev) {
 		return -1;
 	}
 
-	int input = dev->input; // Needs a pointer to int for ioctl()
-	US_LOG_INFO("Using input channel: %d", input);
-	if (_D_XIOCTL(VIDIOC_S_INPUT, &input) < 0) {
-		US_LOG_ERROR("Can't set input channel");
-		return -1;
+	if (dev->capture_type == V4L2_BUF_TYPE_VIDEO_CAPTURE) {
```

Comment:
Multi planar does not supports input selection?

Reply:
I think - yes, they don't support it. Because you can have multi lanes (mplanes) and capture stream from different planes. But i don't have a device with multiple planes, only with 1, so i can't test it on hardware. So in this PR i prefer to stick with 1 plane, until any user with multiplane device will appear.

And one of the first thing i hit after i started debugging was VIDIOC_S_INPUT call.
```
-- INFO  [391395.433      main] -- Starting PiKVM uStreamer 5.43 ...
-- INFO  [391395.433      main] -- Using internal blank placeholder
-- DEBUG [391395.433      main] -- Installing SIGINT handler ...
-- DEBUG [391395.433      main] -- Installing SIGTERM handler ...
-- DEBUG [391395.433      main] -- Ignoring SIGPIPE ...
-- DEBUG [391395.433      main] -- Binding HTTP to [127.0.0.1]:8080 ...
-- INFO  [391395.434      main] -- Listening HTTP on [127.0.0.1]:8080
-- INFO  [391395.434    stream] -- Using V4L2 device: /dev/video0
-- INFO  [391395.434    stream] -- Using desired FPS: 0
-- DEBUG [391395.434    stream] -- _stream_init_loop: stream->run->stop=0
================================================================================
-- INFO  [391395.434    stream] -- Device fd=8 opened
-- INFO  [391395.434      http] -- Starting HTTP eventloop ...
-- DEBUG [391395.435    stream] -- Querying device capabilities ...
-- INFO  [391395.435    stream] -- Using input channel: 0
-- ERROR [391395.435    stream] -- Can't set input channel
-- DEBUG [391395.435    stream] -- Closing device ...
-- INFO  [391395.435    stream] -- Device fd=8 closed
-- INFO  [391395.435    stream] -- Sleeping 1 seconds before new stream init ...
-- DEBUG [391395.451      http] -- HTTP: Updating exposed frame (online=0) ...
-- VERB  [391395.451      http] -- HTTP: Exposed frame: online=0, exp_time=0.000000
```

Reply:
I also have rpi zero 2w with C780 hdmi-to-csi capture and here is `v4l2-ctl --all` output:
```
Driver Info:
	Driver name      : unicam
	Card type        : unicam
	Bus info         : platform:3f801000.csi
	Driver version   : 5.15.68
	Capabilities     : 0x85a00001
		Video Capture
		Metadata Capture
		Read/Write
		Streaming
		Extended Pix Format
		Device Capabilities
	Device Caps      : 0x05200001
		Video Capture
		Read/Write
		Streaming
		Extended Pix Format
Media Driver Info:
	Driver name      : unicam
	Model            : unicam
	Serial           :
	Bus info         : platform:3f801000.csi
	Media version    : 5.15.68
	Hardware revision: 0x00000000 (0)
	Driver version   : 5.15.68
Interface Info:
	ID               : 0x03000005
	Type             : V4L Video
Entity Info:
	ID               : 0x00000003 (3)
	Name             : unicam-image
	Function         : V4L2 I/O
	Flags            : default
	Pad 0x01000004   : 0: Sink
	  Link 0x02000007: from remote pad 0x1000002 of entity 'tc358743 10-000f' (Video Interface Bridge): Data, Enabled, Immutable
Priority: 2
Video input : 0 (Camera 0: ok)
DV timings:
	Active width: 640
	Active height: 480
	Total width: 800
	Total height: 525
	Frame format: progressive
	Polarities: -vsync -hsync
	Pixelclock: 25175000 Hz (59.94 frames per second)
	Horizontal frontporch: 16
	Horizontal sync: 96
	Horizontal backporch: 48
	Vertical frontporch: 10
	Vertical sync: 2
	Vertical backporch: 33
	Standards: CTA-861, DMT
	CTA-861 VIC: 1
	Flags: has CTA-861 VIC
DV timings capabilities:
	Minimum Width: 640
	Maximum Width: 1920
	Minimum Height: 350
	Maximum Height: 1200
	Minimum PClock: 13000000
	Maximum PClock: 165000000
	Standards: CTA-861, DMT, CVT, GTF
	Capabilities: Progressive, Reduced Blanking, Custom Formats
Format Video Capture:
	Width/Height      : 640/480
	Pixel Format      : 'RGB3' (24-bit RGB 8-8-8)
	Field             : None
	Bytes per Line    : 1920
	Size Image        : 921600
	Colorspace        : sRGB
	Transfer Function : Default (maps to sRGB)
	YCbCr/HSV Encoding: Default (maps to ITU-R 601)
	Quantization      : Default (maps to Full Range)
	Flags             :

User Controls

            audio_sampling_rate 0x00981980 (int)    : min=0 max=768000 step=1 default=0 value=48000 flags=read-only
                  audio_present 0x00981981 (bool)   : default=0 value=1 flags=read-only

Digital Video Controls

                  power_present 0x00a00964 (bitmask): max=0x00000001 default=0x00000000 value=1 flags=read-only
```

As you can see here is present:
```
Video input : 0 (Camera 0: ok)
```
So i guess - yes, multiplanar doesn't support VIDIOC_S_INPUT call.

Reply:
Sounds reasonable.